src/nti/contentfragments/tests/test_html.py

```diff
@@ -176,6 +176,12 @@ def test_allowed_attribute_provider(self):
             for attr_name in allowed_attrs:
                 self._test_allowed_attribute_provider(attr_name)
 
+    def test_existing_links(self):
+        # Ensure we properly handle html with existing anchors
+        html = '<html><body><p><a href="http://www.google.com">www.google.com</a></p></body></html>'
```

Comment:
I'd also like this to have a link in a non-standard 'data' attribute, like in the motivating example.

Reply:
If by that you mean an additional attribute like `data-nti-entity-href="http://www.google.com"`, that'll just get stripped out.  I suppose I could add it to an allowed attribute provider, if that's what you were referring to.

Reply:
I do mean exactly that. The point being to demonstrate that the 'find links in text' stuff isn't happening there and expanding in a weird, unexpected place. I could imagine that if it weren't stripped,  etc, this might not be sufficient. Defense against future changes.

Reply:
Fair enough.